### PR TITLE
CRM-16395: Installation with localized default settings (4)

### PR DESCRIPTION
**Pull request by mlutfy**
This is a rebased version of #8538 by @samuelsov.

"New patch for CRM-16395 that replace #7322
Done the new way as proposed by @totten with `SystemInstallEvent` event.
Next step is to create the l10n/LANG/settings.default.json for each language in l10n

To test, copy the file provided [settings.default.json.txt](https://github.com/civicrm/civicrm-core/files/305643/settings.default.json.txt) and copy it as l10n/YOUR_LOCALE/settings.default.json
 and change the values as needed before starting the installation. In the installation process, choose the locale corresponding to the file."

---

 * [CRM-16395: Installation with localized default settings](https://issues.civicrm.org/jira/browse/CRM-16395)

### CODE REVIEW

**Comment by mlutfy** [comment]
I guess I should have done a proper rebase.. in case anyone wants a clean patch:

https://www.bidon.ca/sites/www.bidon.ca/files/tmp/crm-16395.patch

**Comment by mlutfy** [comment]
I tested:

- the normal installation (in US-English)
- installing directly to fr_CA with a l10n/fr_CA/settings.default.json
- poking around on an existing site

However:

- I could not make it work with Drush, but that is out of the current scope and can be a follow-up issue.

It's also worth noting that this patch doesn't do anything unless there is a "l10n/xx_YY/settings.default.json", and currently the l10n package doesn't provide any, so this seems like a fairly low risk patch.

File: CRM/Core/I18n.php [hunk 1]
```diff
@@ -163,14 +163,22 @@ public static function languages($justEnabled = FALSE) {
     if (!$all) {
       $all = CRM_Contact_BAO_Contact::buildOptions('preferred_language');
 
+      // get labels
+      $rows = array();
+      $labels = array();
+      CRM_Core_OptionValue::getValues(array('name' => 'languages'), $rows);
```
Comment (totten) [inline CRM/Core/I18n.php]: @mlutfy @samuelsov In interregnum between the original PR and this revised submission, I think this function was deprecated. On my local dev system, it's showing more warnings like this:

```
Jun 20 00:02:43  [warning] Deprecated function, use CRM_Core_PseudoConstant::getKey
Array
(
    [civi.tag] => deprecated
)
```

Comment (mlutfy) [inline CRM/Core/I18n.php]: Thanks @totten, I'll do a followup PR.

**Pull request merged**